salt/modules/file.py

```diff
@@ -3795,6 +3809,17 @@ def is_link(path):
     # therefore a custom function will need to be called. This function
     # therefore helps API consistency by providing a single function to call for
     # both operating systems.
+    if nostat:
+        if not os.path.exists(path):
```

Comment:
```suggestion
```

This initial if statement should be removed otherwise the function returns incorrect outputs for broken symlinks as it follows the symlink.

```
$ ls -l /usr/directory/broken-symlink /usr/directory/non-existant-target
ls: cannot access '/usr/directory/non-existant-target': No such file or directory 
lrwxrwxrwx 1 user users 34 Feb 22 08:56 /usr/directory/broken-symlink -> /usr/directory/non-existant-target

# test output differs
/usr/directory/broken-symlink
Method 1 (as currently written, including `os.path.exists()` check)
False
Method 2
True
Method 3
True

```

I removed the `if os.path.exists()` check from my local checkout and tested to confirm that the function now returns the same output as the existing functions:
```
/usr/directory/symlink
Method 1
True
Method 2
True
Method 3
True

/usr/bigdirectory/symlink
Method 1
True
Method 2
True
Method 3
True

/usr/directory
Method 1
False
Method 2
False
Method 3
False

/usr/bigdirectory
Method 1
False
Method 2
False
Method 3
False

/usr/directory/broken-symlink
Method 1
True
Method 2
True
Method 3
True

/usr/bigdirectory/broken-symlink
Method 1
True
Method 2
True
Method 3
True

/usr/directory/target
Method 1
False
Method 2
False
Method 3
False

/usr/bigdirectory/target
Method 1
False
Method 2
False
Method 3
False

/usr/directory/non-existant-target
Method 1
False
Method 2
False
Method 3
False

/usr/bigdirectory/non-existant-target
Method 1
False
Method 2
False
Method 3
False

/mnt/mountpoint
Method 1
False
Method 2
False
Method 3
False

/mnt/mountpoint/
Method 1
False
Method 2
False
Method 3
False

/symlink-to-mountpoint
Method 1
True
Method 2
True
Method 3
True

/mnt/mountpoint/subdirectory
Method 1
False
Method 2
False
Method 3
False

/
Method 1
False
Method 2
False
Method 3
False
```